packages/jsxstyle/jsxstyle.tsx

```diff
@@ -1,21 +1,30 @@
 import * as React from 'react';
 import { componentStyles, getStyleCache } from 'jsxstyle-utils';
+import { StyleCache } from 'jsxstyle-utils/src/getStyleCache';
 
-export const cache = getStyleCache();
+export const cache: StyleCache = getStyleCache();
 
-function factory(displayName, defaultProps, tagName) {
-  tagName = tagName || 'div';
-  if (
-    process.env.NODE_ENV === 'development' &&
-    (typeof displayName !== 'string' || !displayName)
-  ) {
-    throw new Error(
-      'makeReactStyleComponentClass expects param 1 to be a valid displayName'
-    );
-  }
+export interface JsxstyleProps {
+  className?: string;
+  component?: string | React.ComponentClass | React.SFC;
+  mediaQueries?: { [key: string]: string };
```

Comment:
Enumerate all possible media queries here if possible.

Reply:
The main problem here is that the media query key can be anything the user specifies. The API does what it needs to do but I’m not super happy with how open-ended it makes things.